### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by palering**
Added "aes-ctr","aes-cfb","aes-ofb" & padding method

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,14 @@
 vendor
 *debug*
-makefile
\ No newline at end of file
+<<<<<<< HEAD
```
Comment (wziww) [inline .gitignore]: Fix conflict first

File: .idea/medusa.iml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (wziww) [inline .idea/medusa.iml]: Do not commit the useless files ( .idea/*

File: conf.json [hunk 3]
```diff
@@ -1,18 +1,19 @@
 {
   "base": {
     "password": "AES256Key-32Characters1234567890",
-    "crypto": "aes-128-gcm"
+    "crypto": "aes-256-gcm",
```
Comment (wziww) [inline conf.json:4]: Change this file unless there is new config settings.
The old config settings should not be  changed.

File: encrpt/aes_gcm_test.go [hunk 4]
```diff
@@ -2,18 +2,13 @@ package encrpt
 
 import (
 	"testing"
-	"github/wziww/medusa/config"
+	// "github/wziww/medusa/config"
```
Comment (wziww) [inline encrpt/aes_gcm_test.go]: Useless code should be removed

File: .gitignore [hunk 5]
```diff
@@ -3,12 +3,5 @@ vendor
 <<<<<<< HEAD
 <<<<<<< HEAD
```
Comment (wziww) [inline .gitignore]: Check this file again plz

Comment (palering) [reply .gitignore]: https://github.com/xiaoxiaoking/medusa/commit/de1a3fdfdb9877d2c8de4b6c7e14628db240b8ec

File: conf.json (hunk 3 above)
Comment (palering) [reply conf.json:4]: https://github.com/xiaoxiaoking/medusa/commit/de1a3fdfdb9877d2c8de4b6c7e14628db240b8ec

File: encrpt/aes_gcm_test.go (hunk 4 above)
Comment (palering) [reply encrpt/aes_gcm_test.go]: https://github.com/xiaoxiaoking/medusa/commit/de1a3fdfdb9877d2c8de4b6c7e14628db240b8ec

File: encrpt/aes_ctr.go [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+// Encode ...
+func (st *AesCtr) Encode(plainBuf []byte) []byte {
+	// pad
+	plainBuf = HandlePadding(st.PaddingMode)(plainBuf, aes.BlockSize)
+	cipherBuf := make([]byte, aes.BlockSize+len(plainBuf))
+	iv := cipherBuf[:aes.BlockSize]
+	if _, err := io.ReadFull(rand.Reader, iv); err != nil {
+		log.FMTLog(log.LOGDEBUG, err)
+		// return nil
```
Comment (wziww) [inline encrpt/aes_ctr.go]: Some where return nil but some where not, which is correct?

File: encrpt/aes_ofb_test.go [hunk 7]
```diff
@@ -0,0 +1,191 @@
+package encrpt
+
+import (
+	"testing"
```
Comment (wziww) [inline encrpt/aes_ofb_test.go]: Check all the test file.
We should init config module first,cause the encrypt module depends on the log module,which depends on the config module.

**Review by wziww — COMMENTED** [review]
Some little issue need to check out.

**Pull request merged**